### DISCUSSION TITLE
Any way to cut out bars?

### DISCUSSION THREAD

**Discussion by Djard** [hydrogen-music/hydrogen#1947] [Hydrogen on Windows]
I can delete bars but doing so leaves a series of silent bars. Is there a way to actually cut out a section: e.g. if bars 40 to 50 are cut out, bar 51 then becomes bar 40?

**Comment by cme** [comment]
In the Song Editor you can select everything to the right of bar 50, then drag them leftward to bar 40. It's clunky, but works. An "Insert" or "Delete" command would be much nicer.

**Reply by Djard** [reply]
Thanks for the clever workaround. The manual states Ctrl+X performs a cut but it is actually copy, which had me thinking I was not following the instruction properly. Again, thanks for the valuable info.

**Comment by cme** [comment]
> The manual states Ctrl+X performs a cut but it is actually copy,

Interesting, that sounds like a bug, could you go into more detail? Ctrl-X should remove selected cells from the song sequence and place them in the cut buffer, and at least does so in macOS and Linux. Thanks!

**Comment by Djard** [comment]
Ctrl+X in Win 10 Pro de-activates cells but does not delete (remove) them, at least not on my PC.
Gman


    On Friday, March 8, 2024 at 03:20:42 PM EST, Colin McEwan ***@***.***> wrote:  
 
 



The manual states Ctrl+X performs a cut but it is actually copy,


Interesting, that sounds like a bug, could you go into more detail? Ctrl-X should remove selected cells from the song sequence and place them in the cut buffer, and at least does so in macOS and Linux. Thanks!

**Reply by theGreatWhiteShark** [reply]
> Ctrl+X in Win 10 Pro de-activates cells but does not delete (remove) them, at least not on my PC.

Yes. This is the intended behavior. <kbd>Ctrl</kbd> + <kbd>X</kbd>|<kbd>C</kbd>|<kbd>V</kbd> do act on the **content** of a cell. Not the cell itself.

It is inspired by spreadsheet handling. And the main reason we went for this approach was to avoid that edits in e.g. the beginning of the song have (potentially unwanted) side effects on following parts.

**Comment by cme** [comment]
> > Ctrl+X in Win 10 Pro de-activates cells but does not delete (remove) them, at least not on my PC.
> 
> Yes. This is the intended behavior. Ctrl + X|C|V do act on the **content** of a cell. Not the cell itself.
> 
> It is inspired by spreadsheet handling. And the main reason we went for this approach was to avoid that edits in e.g. the beginning of the song have (potentially unwanted) side effects on following parts.

Ah! Thanks @theGreatWhiteShark , I was having trouble parsing that. Yeah, the grid of 'empty' cells is more like *the space the patterns exist in* than objects in their own right. Similar also to the grid in a drawing program. Perhaps the appearance of the song editor is misleading, making it look like the empty cells are more tangible and solid than they are? 

I've tried changing the grid lines to dotted and I think this appearance makes it clearer that 'empty' cells are just space, and is less cluttered. Thoughts, @theGreatWhiteShark ?

<img width="608" alt="image" src="https://github.com/hydrogen-music/hydrogen/assets/768466/932cef14-10c3-45e2-9cdf-4bab661aabdc">

**Reply by theGreatWhiteShark** [reply]
> I've tried changing the grid lines to dotted and I think this appearance makes it clearer that 'empty' cells are just space, and is less cluttered.
Thoughts, @theGreatWhiteShark ?

I love it! How about adding it to the 1.2.4 release?

**Reply by cme** [reply]
Smallest PR ever: #1949 😆
